src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -113,6 +113,9 @@
     // A set of methods that result in dialog creations.
     protected static final Set<String> dialogCreatingMethods = new HashSet<String>();
 
+    private transient Set<SIPDialogInjectionListener> dialogInjectionListeners = 
```

Comment:
Because only the first match is taken, you can use the order to priorize sources to inject a dialog.

The first match behavior is choosen, because if we have 2 sources that have the same dialog, there is no criteria, which one we should use. Also first match is the best option to make the injection not longer than really needed.

A HashSet is not really needed, a list can do the same work if lists a prefered